src/postgres/abstract/AbstractPostgresClient.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { PostgreSQLManagementClient } from "@azure/arm-postgresql";
+import { PostgreSQLFlexibleManagementClient } from "@azure/arm-postgresql-flexible";
+import * as msRest from "@azure/ms-rest-js";
+import { IMinimumServiceClientOptions } from "vscode-azureextensionui";
+import { IAbstractPostgresClient } from "./IAbstractPostgresClient";
+import { asAbstractDatabase, asFlexibleParameters, asSingleParameters, flexibleAsAbstractServer, singleAsAbstractServer } from "./maps";
+import * as Models from "./models";
+
+export class AbstractPostgresClient implements IAbstractPostgresClient  {
```

Comment:
I'd be interested to hear what @microsoft/vscodeazuretoolsdev thinks about this but I think we can get rid of `IAbstractPostgresClient` and let this class stand on its own. The interface is useful if we're creating multiple classes that implement it but here it's only used with a single class. And the only other places where it's used are typings for `createAzureClient` calls which can all be replaced with type `AbstractPostgresClient`

Reply:
I agree with Will on this. Right now, IAbstractPostgresClient is only used for declaring type. I think it is a good idea to remove it and just use AbstractPostgresClient, I don't see any harm in it.